src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj

```diff
@@ -63,12 +63,11 @@
   <!-- Packages that are in-box for .NET Core, so we only need to reference them for .NET Framework -->
   <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(DotNetBuildFromSource)' != 'true'">
     <!--
-        MSBuild depends on a new version of SRM/SCI; HostModel a very old one.
-        In order to keep working with MSBuild 17.2, which had binding redirects
-        for SCI 0-5.0.0.0 but not 6, this task must compile against 5, even
-        though that's not the very latest. -->
-    <PackageReference Include="System.Collections.Immutable" Version="5.0.0" />
-    <PackageReference Include="System.Reflection.Metadata" Version="5.0.0" />
+        MSBuild depends on a new version of SRM/SCI; keep in sync with binding redirect:
+          https://github.com/dotnet/msbuild/blob/5d102ae37163fc1f22610bb433c3ab9a2fff91f0/src/MSBuild/app.config#L59
+    -->
+    <PackageReference Include="System.Collections.Immutable" Version="$(SystemCollectionsImmutableVersion)" />
+    <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataVersion)" />
```

Comment:
I was on vacation; what was the specific motivation/break fix for this change?

I think it's ok to go forward with this because we shouldn't care about 17.2 any more, and as long as this moves _ahead_ of MSBuild there should be no problem with adopting a higher version (as long as the task ships both of these assemblies). The `MSBuild.exe` regression tests should catch a problem if there is one.

Reply:
We were trying to get rid of some old pinned dependencies in dotnet/runtime. The Microsoft.NET.HostModel library was referencing an old version of System.Reflection.Metadata to avoid a break that happened 2 years ago, so we were trying to upgrade it as we felt the break was likely fixed. This particular change is undoing a fix that the SDK repo did to mitigate an issue that came up because the Microsoft.NET.HostModel library was referencing an old version of System.Reflection.Metadata (which as mentioned above, was to avoid an earlier break in the sdk).

Reply:
Then I would recommend deleting the comment entirely; they shouldn't be special any more.

Reply:
Looks like the MSBuild binding redirects are actually a little out of date again. They need to be updated for the System.Reflection.Metadata 6.0.1 package (new upper bound assembly version needs to be 6.0.0.1).

In the meantime, (to unblock this PR), I'm going to try to pin this to reference 6.0.0.

To avoid keeping this bad state around, I'll follow up with a PR to MSBuild to update the app.config's binding redirects.

Reply:
I've opened a PR to MSBuild with the updated redirect: https://github.com/dotnet/msbuild/pull/7904

Reply:
Why do the MSBuild binding redirects need to match the reference in the task here from SDK? Are the SDK tasks not delivering SRM.dll?